src/lib.rs

```diff
@@ -481,25 +481,50 @@ pub struct ComparisonInterpreter<T, U> {
     second: U,
 }
 
+/// Creates an `interpreter::SignatureChecker`. The actual type depends on whether the
+/// `signature-validation` feature is enabled. If it’s not, it will create a checker that always
+/// fails.
+#[cfg(feature = "signature-validation")]
+pub fn create_signature_checker(
```

Comment:
This is kind-of missing the point of my [original comment](https://github.com/ZcashFoundation/zcash_script/pull/262#discussion_r2367837703), which was that the semantics of an API (rather than its presence) shouldn't normally be dependent on a feature flag.

Reply:
Yeah, this was me trying to keep the tests intact, when really I should have gone the other way. `create_signature_checker` is now gone. `cxx_rust_comparison_interpreter` is conditionalized on the `signature-validation` feature (because they’re really not comparable if signature validation is disabled), and I’ve split a bunch of tests up that previously used the comparison interpreter to now just test the individual interpreters (the comparison interpreter is exercised by the arbitrary scripts tests).

Reply:
I think this change addresses most of the comments.